src/activity_item_handling.cpp

```diff
@@ -2974,58 +2972,50 @@ void find_auto_consume( player &p, const bool food )
         for( item &it : food_there ) {
             item &comest = p.get_consumable_from( it );
             if( comest.is_null() || comest.is_craft() || !comest.is_food() ||
-                comest.get_comestible_fun() < -5 ) {
+                p.fun_for( comest ).first < -5 ) {
                 // not good eatings.
                 continue;
             }
             if( !p.can_consume( it ) ) {
                 continue;
             }
-            if( comest.rotten() && !saprophage && !saprovore ) {
-                // its gone off.
-                continue;
-            }
-            if( food && comest.get_comestible()->default_nutrition.kcal < 50 ) {
+            if( food && p.compute_effective_nutrients( comest ).kcal < 50 ) {
                 // not filling enough
                 continue;
             }
-            if( ( p.allergy_type( comest ) != MORALE_NULL ) || ( carnivore &&
-                    comest.has_flag( "ALLERGEN_JUNK" ) &&
-                    !comest.has_flag( "CARNIVORE_OK" ) ) ) {
-                // allergic to.
+            if( !p.will_eat( comest, false ).success() ) {
+                // wont like it, cannibal meat etc
                 continue;
             }
-            if( comest.has_flag( "CANNIBALISM" ) && !cannibal ) {
-                // it's people.
+            if( !it.is_owned_by( p, true ) ) {
+                // it aint ours.
                 continue;
             }
             if( !food && comest.get_comestible()->quench < 15 ) {
                 // not quenching enough
                 continue;
             }
-            if( comest.item_tags.count( "FROZEN" ) ) {
-                // its frozen
-                continue;
-            }
-            if( comest.charges > 0 && p.stomach.stomach_remaining( p ) < comest.volume() / comest.charges &&
-                !comest.has_infinite_charges() ) {
-                // too full to eat that.
+            if( comest.item_tags.count( "FROZEN" ) && !comest.has_flag( flag_MELTS ) &&
```

Comment:
This is already checked in `player::will_eat` (in the `player::can_eat` call, to be precise).